File: src/irrgarten/LabyrinthCharacter.java
package irrgarten;

public abstract class LabyrinthCharacter {
    private String name;
    private float intelligence;
    private float strength;
    private float health;
    private Integer row;
    private Integer col;

    LabyrinthCharacter(String name, float intelligence , float strength,float health){
        this.name = name;
        this.intelligence=intelligence;
        this.strength=strength;
        this.health=health;
        this.row=0;
        this.col=0;
    }
    LabyrinthCharacter(LabyrinthCharacter other){
        this(other.name,other.intelligence,other.strength,other.health);
        this.col=other.col;
        this.row=other.row;

        
    }
    public boolean dead(){
        if (this.health<=0){
            return true;
        }
        else{
            return false;
        }
    }
    public int getRow(){
        return this.row;
    }

    public int getCol(){
        return this.col;
    }
    protected float getIntellignece(){
        return this.intelligence;
    }
    protected float getStrength(){
        return this.strength;
    }
    protected float getHealth(){
        return this.health;
    }
    protected void setHealth(float health){
        this.health=health;
    }
    public void setPos(Integer row, Integer col){
        this.row=row;
        this.col=col;
    }
    @Override
    public String toString() {
        return String.format("[%s,%f,%f,%f]",this.name,this.intelligence, this.strength, this.health);
    }
    protected void gotWounded(){
        this.health --;
    }
    public abstract float attack();
    public abstract boolean defend(float attack);

}

File: JUEGO-RUBY/irrgarten/LabyrinthCharacter.rb
module Irrgarten
    class LabyrinthCharacter
        attr_reader :intelligence,:row, :col, :intelligence, :strenght ,:name
        attr_accessor :health
        def initialize(name,intelligence,strenght,health)
            @name = name
            @intelligence = intelligence
            @strenght = strenght
            @health = health
            @row=0
            @col=0
        end
        def copy(other)
            @name = other.name
            @intelligence = other.intelligence
            @strenght = other.strenght
            @health = other.health
            @col = other.col
            @row = other.row

        end
        #Devuelve true si el jugador esta muerto
        def dead
            return @health <=0
        end
        #Establece la posicion del jugador 
        def set_pos(row , col)
            @row = row
            @col = col 
        end
        #Muestra los atributos del jugador en formato string
        def to_s
            return "[#@name,H:#@health,I:#@intelligence,S:#@strenght,pos(#@row,#@col)]"
        end
        #decrementa en 1 la salud del jugador 
        def got_wounded
            if @health >1
                @health -=1
            else
                @health = 0
            end
        end


    end
end
